src/registry/children/validationSchema.spec.ts

```diff
@@ -0,0 +1,232 @@
+import type {ChildrenComponentSchema} from '@open-formulieren/types';
+import {addYears, subDays, subYears} from 'date-fns';
+import {createIntl} from 'react-intl';
+import {describe, expect, test} from 'vitest';
+
+import {getRegistryEntry} from '@/registry/registry';
+
+import getValidationSchema from './validationSchema';
+
+const intl = createIntl({locale: 'en', messages: {}});
+
+const BASE_COMPONENT: ChildrenComponentSchema = {
+  type: 'children',
+  id: 'children',
+  key: 'children',
+  label: 'Children',
+  enableSelection: false,
+};
+
+const buildValidationSchema = (component: ChildrenComponentSchema) => {
+  const schemas = getValidationSchema(component, {
+    intl,
+    getRegistryEntry,
+    validatePlugins: async () => undefined,
+  });
+  return schemas[component.key];
+};
+
+const today = new Date();
+
+describe('Plain children validation', () => {
+  test.each([undefined, []])('accepts zero child details', async value => {
+    const schema = buildValidationSchema(BASE_COMPONENT);
+
+    const {success} = await schema.safeParseAsync(value);
+
+    expect(success).toBe(true);
+  });
+
+  test('accepts multiple child details', async () => {
+    const schema = buildValidationSchema(BASE_COMPONENT);
+
+    const {success} = await schema.safeParseAsync([
+      {
+        bsn: '111222333',
+        firstNames: 'Johan',
+        dateOfBirth: '1980-12-12',
+      },
+      {
+        bsn: '923456788',
+        firstNames: 'Joe',
+        dateOfBirth: '1980-12-12',
+      },
+    ]);
+
+    expect(success).toBe(true);
+  });
+});
+
+describe('Children subfields validation', () => {
+  test.each([
+    [{bsn: '', firstNames: '', dateOfBirth: ''}],
+
+    [{bsn: '111222333', firstNames: '', dateOfBirth: ''}],
+    [{bsn: '111222333', firstNames: '', dateOfBirth: '1980-12-12'}],
+    [{bsn: '111222333', firstNames: 'Joe', dateOfBirth: ''}],
+
+    [{bsn: '', firstNames: 'Joe', dateOfBirth: ''}],
+    [{bsn: '', firstNames: 'Joe', dateOfBirth: '1980-12-12'}],
+
+    [{bsn: '', firstNames: '', dateOfBirth: '1980-12-12'}],
+  ])(
+    'does not accept children with missing BSN, first name or date of birth (value: %s)',
+    async (...value) => {
```

Comment:
the spread syntax is tripping me up here. Can't this be:


```suggestion
  test.each([
    [{bsn: '', firstNames: '', dateOfBirth: ''}],

    [{bsn: '111222333', firstNames: '', dateOfBirth: ''}],
    [{bsn: '111222333', firstNames: '', dateOfBirth: '1980-12-12'}],
    [{bsn: '111222333', firstNames: 'Joe', dateOfBirth: ''}],

    [{bsn: '', firstNames: 'Joe', dateOfBirth: ''}],
    [{bsn: '', firstNames: 'Joe', dateOfBirth: '1980-12-12'}],

    [{bsn: '', firstNames: '', dateOfBirth: '1980-12-12'}],
  ])(
    'does not accept children with missing BSN, first name or date of birth (value: %s)',
    async (value: ChildDetails[]) => {
```